### PR TITLE
HDDS-1551. Implement Bucket Write Requests to use Cache and DoubleBuffer.

### PR DESCRIPTION
**Pull request by bharatviswa504**

### CODE REVIEW

**Comment by bharatviswa504** [comment]
This is dependent on HDDS-1499 and HDDS-1512.

The last commit is part of this Jira. Opened a PR to get a Jenkins run and to get initial comments on the class design approach, so that similar can be followed for other requests.

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java [hunk 1]
```diff
@@ -313,6 +313,8 @@ private OzoneManager(OzoneConfiguration conf) throws IOException,
     RPC.setProtocolEngine(configuration, OzoneManagerProtocolPB.class,
         ProtobufRpcEngine.class);
 
+    metadataManager = new OmMetadataManagerImpl(configuration);
+
```
Comment (hadoop-yetus) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java]: whitespace:end of line

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 36 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 10 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 76 | Maven dependency ordering for branch |
| +1 | mvninstall | 421 | trunk passed |
| +1 | compile | 196 | trunk passed |
| +1 | checkstyle | 52 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 877 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 121 | trunk passed |
| 0 | spotbugs | 241 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 423 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 23 | Maven dependency ordering for patch |
| +1 | mvninstall | 395 | the patch passed |
| +1 | compile | 202 | the patch passed |
| +1 | cc | 202 | the patch passed |
| +1 | javac | 202 | the patch passed |
| +1 | checkstyle | 55 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| -1 | whitespace | 0 | The patch has 1 line(s) that end in whitespace. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply |
| +1 | shadedclient | 721 | patch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 60 | hadoop-ozone generated 4 new + 2 unchanged - 0 fixed = 6 total (was 2) |
| +1 | findbugs | 456 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 159 | hadoop-hdds in the patch failed. |
| -1 | unit | 226 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 37 | The patch does not generate ASF License warnings. |
| | | 4726 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.om.ratis.TestOzoneManagerRatisServer |
|   | hadoop.ozone.om.ratis.TestOzoneManagerStateMachine |
|   | hadoop.ozone.security.TestOzoneDelegationTokenSecretManager |
|   | hadoop.ozone.om.TestOzoneManagerLock |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-827/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/827 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle cc |
| uname | Linux 1ebd0979f400 4.4.0-143-generic #169~14.04.2-Ubuntu SMP Wed Feb 13 15:00:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / c183bd8 |
| Default Java | 1.8.0_212 |
| whitespace | https://builds.apache.org/job/hadoop-multibranch/job/PR-827/2/artifact/out/whitespace-eol.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-827/2/artifact/out/diff-javadoc-javadoc-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-827/2/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-827/2/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-827/2/testReport/ |
| Max. process+thread count | 1187 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common hadoop-ozone/common hadoop-ozone/integration-test hadoop-ozone/ozone-manager U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-827/2/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java [hunk 2]
```diff
@@ -313,6 +313,7 @@ private OzoneManager(OzoneConfiguration conf) throws IOException,
     RPC.setProtocolEngine(configuration, OzoneManagerProtocolPB.class,
         ProtobufRpcEngine.class);
 
+    metadataManager = new OmMetadataManagerImpl(configuration);
```
Comment (hadoop-yetus) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java]: whitespace:end of line

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 25 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 13 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 72 | Maven dependency ordering for branch |
| +1 | mvninstall | 415 | trunk passed |
| +1 | compile | 195 | trunk passed |
| +1 | checkstyle | 52 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 878 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 121 | trunk passed |
| 0 | spotbugs | 251 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 434 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 59 | Maven dependency ordering for patch |
| +1 | mvninstall | 392 | the patch passed |
| +1 | compile | 205 | the patch passed |
| +1 | cc | 205 | the patch passed |
| +1 | javac | 205 | the patch passed |
| +1 | checkstyle | 56 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| -1 | whitespace | 0 | The patch has 1 line(s) that end in whitespace. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply |
| +1 | shadedclient | 711 | patch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 62 | hadoop-ozone generated 4 new + 2 unchanged - 0 fixed = 6 total (was 2) |
| +1 | findbugs | 440 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 148 | hadoop-hdds in the patch failed. |
| -1 | unit | 1101 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 33 | The patch does not generate ASF License warnings. |
| | | 5546 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-827/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/827 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle cc |
| uname | Linux 68eb0f62efa3 4.4.0-139-generic #165~14.04.1-Ubuntu SMP Wed Oct 31 10:55:11 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 12c8161 |
| Default Java | 1.8.0_212 |
| whitespace | https://builds.apache.org/job/hadoop-multibranch/job/PR-827/4/artifact/out/whitespace-eol.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-827/4/artifact/out/diff-javadoc-javadoc-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-827/4/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-827/4/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-827/4/testReport/ |
| Max. process+thread count | 3972 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common hadoop-ozone/common hadoop-ozone/integration-test hadoop-ozone/ozone-manager U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-827/4/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 30 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 10 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 46 | Maven dependency ordering for branch |
| +1 | mvninstall | 562 | trunk passed |
| +1 | compile | 276 | trunk passed |
| +1 | checkstyle | 73 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 855 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 160 | trunk passed |
| 0 | spotbugs | 290 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 478 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 21 | Maven dependency ordering for patch |
| +1 | mvninstall | 551 | the patch passed |
| +1 | compile | 269 | the patch passed |
| +1 | cc | 269 | the patch passed |
| +1 | javac | 269 | the patch passed |
| +1 | checkstyle | 76 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 632 | patch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 83 | hadoop-ozone generated 4 new + 2 unchanged - 0 fixed = 6 total (was 2) |
| +1 | findbugs | 499 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 163 | hadoop-hdds in the patch failed. |
| -1 | unit | 1409 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 61 | The patch does not generate ASF License warnings. |
| | | 9664 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdds.scm.pipeline.TestRatisPipelineProvider |
|   | hadoop.ozone.container.common.impl.TestContainerPersistence |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-827/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/827 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle cc |
| uname | Linux 7f4c042b294b 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 9c61494 |
| Default Java | 1.8.0_212 |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-827/5/artifact/out/diff-javadoc-javadoc-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-827/5/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-827/5/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-827/5/testReport/ |
| Max. process+thread count | 5344 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/common hadoop-ozone/ozone-manager hadoop-ozone/integration-test U: hadoop-ozone |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-827/5/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 30 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 10 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 12 | Maven dependency ordering for branch |
| +1 | mvninstall | 527 | trunk passed |
| +1 | compile | 268 | trunk passed |
| +1 | checkstyle | 85 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 939 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 148 | trunk passed |
| 0 | spotbugs | 335 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 525 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 19 | Maven dependency ordering for patch |
| +1 | mvninstall | 485 | the patch passed |
| +1 | compile | 283 | the patch passed |
| +1 | cc | 283 | the patch passed |
| +1 | javac | 283 | the patch passed |
| -0 | checkstyle | 40 | hadoop-ozone: The patch generated 8 new + 0 unchanged - 0 fixed = 8 total (was 0) |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 738 | patch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 89 | hadoop-ozone generated 5 new + 2 unchanged - 0 fixed = 7 total (was 2) |
| +1 | findbugs | 517 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 159 | hadoop-hdds in the patch failed. |
| -1 | unit | 1139 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 46 | The patch does not generate ASF License warnings. |
| | | 9471 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.client.rpc.TestWatchForCommit |
|   | hadoop.ozone.ozShell.TestOzoneShell |
|   | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |
|   | hadoop.ozone.om.TestOmMetrics |
|   | hadoop.ozone.client.rpc.TestOzoneAtRestEncryption |
|   | hadoop.ozone.container.common.impl.TestContainerPersistence |
|   | hadoop.ozone.om.TestOzoneManagerHA |
|   | hadoop.ozone.web.client.TestBuckets |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-827/6/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/827 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle cc |
| uname | Linux d9ebc5853ef6 4.4.0-139-generic #165~14.04.1-Ubuntu SMP Wed Oct 31 10:55:11 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 9c61494 |
| Default Java | 1.8.0_212 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-827/6/artifact/out/diff-checkstyle-hadoop-ozone.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-827/6/artifact/out/diff-javadoc-javadoc-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-827/6/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-827/6/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-827/6/testReport/ |
| Max. process+thread count | 4868 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/common hadoop-ozone/ozone-manager hadoop-ozone/integration-test U: hadoop-ozone |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-827/6/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 32 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 10 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 87 | Maven dependency ordering for branch |
| +1 | mvninstall | 587 | trunk passed |
| +1 | compile | 281 | trunk passed |
| +1 | checkstyle | 84 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 953 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 152 | trunk passed |
| 0 | spotbugs | 306 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 512 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 30 | Maven dependency ordering for patch |
| +1 | mvninstall | 472 | the patch passed |
| +1 | compile | 277 | the patch passed |
| +1 | cc | 277 | the patch passed |
| +1 | javac | 277 | the patch passed |
| +1 | checkstyle | 78 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 705 | patch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 74 | hadoop-ozone generated 1 new + 2 unchanged - 0 fixed = 3 total (was 2) |
| -1 | findbugs | 214 | hadoop-hdds generated 3 new + 0 unchanged - 0 fixed = 3 total (was 0) |
||| _ Other Tests _ |
| -1 | unit | 173 | hadoop-hdds in the patch failed. |
| -1 | unit | 1170 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 46 | The patch does not generate ASF License warnings. |
| | | 6511 | |


| Reason | Tests |
|-------:|:------|
| FindBugs | module:hadoop-hdds |
|  |  Synchronization performed on java.util.concurrent.ConcurrentHashMap in org.apache.hadoop.utils.db.cache.PartialTableCache.evictCache(long)  At PartialTableCache.java:org.apache.hadoop.utils.db.cache.PartialTableCache.evictCache(long)  At PartialTableCache.java:[line 90] |
|  |  Synchronization performed on java.util.concurrent.ConcurrentHashMap in org.apache.hadoop.utils.db.cache.PartialTableCache.get(CacheKey)  At PartialTableCache.java:org.apache.hadoop.utils.db.cache.PartialTableCache.get(CacheKey)  At PartialTableCache.java:[line 61] |
|  |  Synchronization performed on java.util.concurrent.ConcurrentHashMap in org.apache.hadoop.utils.db.cache.PartialTableCache.put(CacheKey, CacheValue)  At PartialTableCache.java:org.apache.hadoop.utils.db.cache.PartialTableCache.put(CacheKey, CacheValue)  At PartialTableCache.java:[line 68] |
| Failed junit tests | hadoop.ozone.client.rpc.TestOzoneAtRestEncryption |
|   | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |
|   | hadoop.ozone.container.common.impl.TestContainerPersistence |
|   | hadoop.ozone.ozShell.TestOzoneShell |
|   | hadoop.ozone.om.TestOmMetrics |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |
|   | hadoop.ozone.web.client.TestBuckets |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-827/8/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/827 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle cc |
| uname | Linux 9544cfb72bf7 4.4.0-143-generic #169~14.04.2-Ubuntu SMP Wed Feb 13 15:00:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 9c61494 |
| Default Java | 1.8.0_212 |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-827/8/artifact/out/diff-javadoc-javadoc-hadoop-ozone.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-827/8/artifact/out/new-findbugs-hadoop-hdds.html |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-827/8/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-827/8/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-827/8/testReport/ |
| Max. process+thread count | 4370 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common hadoop-ozone/common hadoop-ozone/integration-test hadoop-ozone/ozone-manager U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-827/8/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 30 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 10 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 65 | Maven dependency ordering for branch |
| +1 | mvninstall | 519 | trunk passed |
| +1 | compile | 279 | trunk passed |
| +1 | checkstyle | 75 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 826 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 146 | trunk passed |
| 0 | spotbugs | 283 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 472 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 28 | Maven dependency ordering for patch |
| +1 | mvninstall | 459 | the patch passed |
| +1 | compile | 254 | the patch passed |
| +1 | cc | 254 | the patch passed |
| +1 | javac | 254 | the patch passed |
| +1 | checkstyle | 71 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 610 | patch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 83 | hadoop-ozone generated 1 new + 2 unchanged - 0 fixed = 3 total (was 2) |
| -1 | findbugs | 212 | hadoop-hdds generated 3 new + 0 unchanged - 0 fixed = 3 total (was 0) |
||| _ Other Tests _ |
| -1 | unit | 150 | hadoop-hdds in the patch failed. |
| -1 | unit | 1100 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 43 | The patch does not generate ASF License warnings. |
| | | 9095 | |


| Reason | Tests |
|-------:|:------|
| FindBugs | module:hadoop-hdds |
|  |  Synchronization performed on java.util.concurrent.ConcurrentHashMap in org.apache.hadoop.utils.db.cache.PartialTableCache.evictCache(long)  At PartialTableCache.java:org.apache.hadoop.utils.db.cache.PartialTableCache.evictCache(long)  At PartialTableCache.java:[line 90] |
|  |  Synchronization performed on java.util.concurrent.ConcurrentHashMap in org.apache.hadoop.utils.db.cache.PartialTableCache.get(CacheKey)  At PartialTableCache.java:org.apache.hadoop.utils.db.cache.PartialTableCache.get(CacheKey)  At PartialTableCache.java:[line 61] |
|  |  Synchronization performed on java.util.concurrent.ConcurrentHashMap in org.apache.hadoop.utils.db.cache.PartialTableCache.put(CacheKey, CacheValue)  At PartialTableCache.java:org.apache.hadoop.utils.db.cache.PartialTableCache.put(CacheKey, CacheValue)  At PartialTableCache.java:[line 68] |
| Failed junit tests | hadoop.ozone.client.rpc.TestBlockOutputStreamWithFailures |
|   | hadoop.ozone.client.rpc.TestWatchForCommit |
|   | hadoop.ozone.container.common.impl.TestContainerPersistence |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-827/7/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/827 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle cc |
| uname | Linux b6c070e2d924 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 9c61494 |
| Default Java | 1.8.0_212 |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-827/7/artifact/out/diff-javadoc-javadoc-hadoop-ozone.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-827/7/artifact/out/new-findbugs-hadoop-hdds.html |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-827/7/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-827/7/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-827/7/testReport/ |
| Max. process+thread count | 4815 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common hadoop-ozone/common hadoop-ozone/integration-test hadoop-ozone/ozone-manager U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-827/7/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by arp7** [comment]
Obsoleted by #850 .

**Pull request closed without merging**